Simplenote/Classes/SPTagsListViewController.m

```diff
@@ -319,125 +262,231 @@ - (void)tableView:(UITableView *)tableView performAction:(SEL)action forRowAtInd
+    BOOL isTagRow = indexPath.section == SPTagsListSectionTags;
+    return self.bEditing && isTagRow ? UITableViewCellEditingStyleDelete : UITableViewCellEditingStyleNone;
 }
 
 - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
-    
     if (editingStyle == UITableViewCellEditingStyleDelete) {
         [SPTracker trackTagRowDeleted];
 		[self removeTagAtIndexPath:indexPath];
     }
 }
 
 
+#pragma mark - Cell Setup
+
+- (void)configureCell:(SPTagListViewCell *)cell atIndexPath:(NSIndexPath *)indexPath {
+    switch (indexPath.section) {
+        case SPTagsListSectionSystem: {
+            [self configureSystemCell:cell atIndexPath:indexPath];
+            break;
+        }
+        case SPTagsListSectionTags: {
+            [self configureTagCell:cell atIndexPath:indexPath];
+            break;
+        }
+    }
+}
+
+- (void)configureSystemCell:(SPTagListViewCell *)cell atIndexPath:(NSIndexPath *)indexPath {
+    switch (indexPath.row) {
+        case SPTagsListSystemRowAllNotes: {
+            cell.tagNameTextField.text = NSLocalizedString(@"All Notes", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameAllNotes];
+            break;
+        }
+        case SPTagsListSystemRowTrash: {
+            cell.tagNameTextField.text = NSLocalizedString(@"Trash-noun", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameTrash];
+            break;
+        }
+        case SPTagsListSystemRowSettings: {
+            cell.tagNameTextField.text = NSLocalizedString(@"Settings", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameSettings];
+            break;
+        }
+    }
+}
+
+- (void)configureTagCell:(SPTagListViewCell *)cell atIndexPath:(NSIndexPath *)indexPath {
+    NSString *tagName = [self tagAtTableViewIndexPath:indexPath].name;
+
+    cell.tagNameTextField.text = tagName;
+    cell.tagNameTextField.delegate = self;
+    cell.leftImage = nil;
+    cell.accessibilityLabel = tagName;
+    cell.delegate = self;
+}
+
+- (BOOL)shouldSelectCellAtIndexPath:(NSIndexPath *)indexPath {
+    NSString *selectedTag = SPAppDelegate.sharedDelegate.selectedTag;
+
+    switch (indexPath.section) {
+        case SPTagsListSectionSystem: {
+            switch (indexPath.row) {
+                case SPTagsListSystemRowAllNotes: {
+                    return selectedTag == nil;
+                }
+                case SPTagsListSystemRowTrash: {
+                    return selectedTag == kSimplenoteTagTrashKey;
+                }
+                case SPTagsListSystemRowSettings: {
+                    return NO;
+                }
+            }
+            break;
+        }
+        case SPTagsListSectionTags: {
+            return selectedTag == [self tagAtTableViewIndexPath:indexPath].name;
+        }
+    }
```

Comment:
Aqui tambien.

Actually, with the nested switch statement they might make sense 🤔.

Reply:
I think only `case SPTagsListSectionSystem` requires the `{ ... }`. On the others cases the value can be returned without {}